src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -513,7 +513,8 @@ public SymbolInfo GetSymbolInfo(ExpressionSyntax expression, CancellationToken c
 
             SyntaxNode parent;
 
-            if (!CanGetSemanticInfo(expression, allowNamedArgumentName: true))
+            if (!CanGetSemanticInfo(expression, allowNamedArgumentName: true) ||
+                expression.Kind() == SyntaxKind.DeclarationExpression) // Declaration doesn't refer to a symbol
```

Comment:
Why not move this check into CanGetSemanticInfo?  
The doc comment on CanGetSemanticInfo specifically says "could be successfully interrogated by GetSymbolInfo/GetTypeInfo/GetMemberGroup/GetConstantValue".